src/constants/MarkdownFiles/posts/2025-07-06-gsoc-25-SafwanSayeed-week05.md

```diff
@@ -0,0 +1,76 @@
+---
+title: "GSoC '25 Week 5 Update by Safwan Sayeed"
+excerpt: "Implementing the Symbol Table and Memory Module Integration"
+category: "DEVELOPER NEWS"
+date: "2025-07-06"
+slug: "2025-07-06-gsoc-25-sa-fw-an-week5"
+author: "@/constants/MarkdownFiles/authors/safwan-sayeed.md"
+tags: "gsoc25,sugarlabs,week5,sa-fw-an"
+image: "assets/Images/GSOC.png"
+---
+
+<!-- markdownlint-disable -->
+
+# Week 5 Progress Report by Safwan Sayeed
+
+**Project:** Music Blocks 4 Program Engine  
+**Mentors:** [Anindya Kundu](https://github.com/meganindya/), [Sumit Srivastava](https://github.com/sum2it)  
+**Assisting Mentors:** [Devin Ullibari](https://github.com/pikurasa/), [Walter Bender](https://github.com/walterbender)  
+**Reporting Period:** 2025-06-30 - 2025-07-06  
+
+---
+
+## A Blog-style Retrospective
+
+This week I implemented the Symbol Table and integrated it with the Memory Module to manage variable scopes and dependencies effectively. This was a crucial step in preparing for the AST-to-IR compilation logic, ensuring that our program engine can handle variable resolution correctly during the translation process.
```

Comment:
Please put what AST stands for at its first instance.

For example:
These three letter acronym (TLA) need to be spelled out the first time we write them.

Subsequent iterations can just use the acronym.

If this is something that was explained in more detail in a previous post, please link to it.